.envrc

```diff
@@ -0,0 +1,208 @@
+        info "You're in a virtualenv, but it's not in the expected location (${PWD}/.venv)"
+        init_venv
+    fi
+else
+    if [ ! -f ".venv/bin/activate" ]; then
+        info "You don't seem to have a virtualenv."
+        init_venv
+    fi
+fi
+
+info "Activating virtualenv."
+source .venv/bin/activate
+[ "$(command -v python)" != "${PWD}/.venv/bin/python" ] && die "Failed to activate virtualenv."
+
+python -c "import sys; sys.exit(sys.version_info[:2] != (2, 7))" || \
+    die "For some reason, the virtualenv isn't Python 2.7."
+
+if [ "$(command -v sentry)" != "${PWD}/.venv/bin/sentry" ]; then
+    info "Your .venv is activated, but sentry doesn't seem to be installed. Let's install it."
+    make ensure-pinned-pip
+    SENTRY_LIGHT_BUILD=1 make install-sentry-dev
+fi
+
+
+### pre-commit ###
+
+info "Checking pre-commit..."
+
+# this is cheap, so we'll just do it every time
+ln -sf config/hooks/* .git/hooks
+
+if ! require pre-commit; then
+    info "Looks like you don't have pre-commit installed. Let's install it."
+    make setup-git
+fi
+
+# this hotfix is cheap too, so just run it every time
+rm -f .git/hooks/pre-commit.legacy
+
+
+### devservices ###
+
+info "Checking devservices..."
+
+# XXX: these container names are hardcoded for now
+# NOTE: sentry_symbolicator isn't started up by devservices, and is behind a config flag, so we're not checking for it
+for container in \
+    sentry_postgres     \
+    sentry_clickhouse   \
+    sentry_snuba        \
+    sentry_redis        ;
+    do
+    docker exec "$container" true || die "The docker container ${container} doesn't seem to be running.
+Please run sentry devservices up."
+done
+
+
+### database ###
+
+info "Checking database..."
+
+if ! docker exec sentry_postgres sh \
+    -c "psql -U postgres -h 127.0.0.1 sentry --command 'select 1 from sentry_useremail'" >/dev/null;
+    then
+    die "It doesn't look like you have a database for sentry yet - you'll need to run 'make reset-db'."
+fi
+
+
+### Node ###
+
+info "Checking node..."
+
+node_version="10.16.3"
+
+# It would be nice to enforce that node is installed via volta (and is therefore a shim that will check against
+# the node pin in package.json), but for now, let's just explicitly check the node version.
+
+if ! require node; then
+    die "You don't seem to have node installed. We want version ${node_version}.
+It's recommended to use volta - please refer to https://docs.sentry.io/development/contribute/environment"
+fi
+
+if [ "$(node -v)" != "v${node_version}" ]; then
+    die "Your node version doesn't match ${node_version}.
+It's recommended to use volta - please refer to https://docs.sentry.io/development/contribute/environment"
+fi
+
+if [ ! -x "node_modules/.bin/webpack" ]; then
+    info "You don't seem to have yarn packages installed. Let's install them."
+    make install-yarn-pkgs
+fi
+
+PATH_add node_modules/.bin
+
+
+### Overrides ###
+
+if [ -f .envrc-overrides ]; then
```

Comment:
This is a risk fwiw. If someone writes to this file it’ll get run without confirmation. That’s why I chose to do .env only.

Reply:
Sure. It's for power users though, I think it's fair to assume they know what they're doing.

Reply:
I’m saying I can root your machine. Doesn’t matter if you’re a power user.

Reply:
I’m saying I can root your machine. Doesn’t matter if you’re a power user.

Reply:
Discussed offline and I was convinced. 16999de